src/lib/Debridge/parser.ts

```diff
@@ -0,0 +1,186 @@
+import { ethers } from 'ethers';
+import { ACTION_ENUM } from '../../enums';
+import { ProtocolHelper } from '../../helpers';
+import { IBridgeInAction, ITransaction, ITransactionAction, ITransactionLog } from '../../types';
+import { CONTRACT_ENUM, contracts, EVENT_ENUM } from './contracts';
+
+
+
+
+export class DlnSourceContractParseTransaction {
+  private static contractDefiniton = contracts[CONTRACT_ENUM.DLN_SOURCE];
+
+  public static parseTransaction(transaction: ITransaction): ITransactionAction[] {
+    const actions: ITransactionAction[] = [];
+
+    const matchedPlacedOrderLog = transaction.logs.find((log) => log.topics[0] === EVENT_ENUM.ORDER_PLACED);
+    if (matchedPlacedOrderLog) {
+      actions.push(this.parsePlacedOrder(matchedPlacedOrderLog));
+    }
+    return actions;
+  }
+
+
+  private static parsePlacedOrder(placeOrderLog: ITransactionLog): IBridgeInAction {
+    try {
+      const parsedLog = ProtocolHelper.parseLog(
+        placeOrderLog,
+        this.contractDefiniton.events[EVENT_ENUM.ORDER_PLACED]
+      );
+
+      console.log(parsedLog);
+    
+      const structuredEvent = this.mapDecodedEvent(parsedLog);
+
+      console.log(structuredEvent);
+      
+      
+      const order = structuredEvent.order;
+
+      return {
+        type: ACTION_ENUM.BRIDGE_IN,
+        fromChain: Number(order.giveChainId),
+        toChain: order.takeChainId.toString(),
+        fromToken: order.giveTokenAddress,
+        toToken: order.takeTokenAddress,
+        fromAmount: (BigInt(order.giveAmount) + BigInt(structuredEvent.percentFee)).toString(),
+        toAmount: order.takeAmount.toString(),
+        sender: order.makerSrc,
+        recipient: order.receiverDst
+      };
+    } catch (error) {
+      console.error('Error parsing order placed transaction:', error);
```

Comment:
Lets not have any console logs in the parser. You can just throw with an appropriate message

Reply:
Done

Reply:
Let's discuss this

Reply:
Done